src/Analysis/Codelyzer.Analysis/CodeGraph.cs

```diff
@@ -8,137 +8,278 @@ namespace Codelyzer.Analysis
+            {
+                if (child != null)
+                {
+                    var currentNode = new Node();
+                    currentNode.ParentNode = parentNode;
+                    currentNode.UstNode = child;
+                    try
+                    {
+                        // These children are the node objects
+                        if (IsNode(child))
+                        {
+                            currentNode.NodeType = GetNodeType(child);
+                            currentNode.Name = child.Identifier;
+                            currentNode.Identifier = child.FullIdentifier;
+                            if (!Graph.Contains(currentNode))
+                            {
+                                childNodes.Add(currentNode);
+                                Graph.Add(currentNode);
+                                currentNode.ChildNodes = InitializeNodesHelper(child, currentNode);
+                            }
+                        }
+                        else
+                        {
+                            // These children are the edges
+                            if (IsEdgeConnection(child))
+                            {
+                                GetOrAddEdgeCandidates(parentNode)?.Add(child);
+                            }
+                            // Keep going until there are no more children
+                            childNodes = childNodes.Union(InitializeNodesHelper(child, parentNode)).ToHashSet();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, "Error while populating children");
+                    }
+                }
+            }
+            return childNodes;
         }
-        private void AddEdges()
+        private void CreateClassHierarchyEdges(Node sourceNode)
         {
-            // Add inheritance relations
-            classDeclarations.ToList().ForEach(classDeclaration =>
+            if (sourceNode.UstNode is ClassDeclaration classDeclaration)
             {
-                var sourceNode = GetClassNode(classDeclaration.FullIdentifier);
-
                 // Check base types list for interfaces
                 var baseTypes = classDeclaration.BaseList;
                 var baseTypeOriginalDefinition = classDeclaration.BaseTypeOriginalDefinition;
-                if(!string.IsNullOrEmpty(baseTypeOriginalDefinition) && baseTypeOriginalDefinition != "object")
+                if (!string.IsNullOrEmpty(baseTypeOriginalDefinition) && baseTypeOriginalDefinition != "object")
                 {
                     baseTypes.Add(baseTypeOriginalDefinition);
                 }
                 baseTypes.ForEach(baseType =>
                 {
-                    var targetNode = GetClassNode(baseType);
+                    var targetNode = ClassNodes.FirstOrDefault(n => n.Identifier == baseType);
```

Comment:
Couldn't this `InterfaceNodes` also contain `baseType`?

Reply:
Good point. Also, interfaces can have a baselist too, I will need to make some changes to the interface declarations to support this.